File: 2025/day06_1/src/main/python/cephalopod_formula.py
from cephalopod_operation import CephalopodOperation


class CephalopodFormula:
    def __init__(self):
        self.params = []
        self.operation = CephalopodOperation.NONE

    def add_param(self, new_param: int):
        self.params.append(new_param)

    def __str__(self) -> str:
        return self.operation.value.join(str(p) for p in self.params)

    def do_homework(self) -> int:
        if not self.params:
            return 0

        result = self.params[0]
        for p in self.params[1:]:
            result = self.operation.calculate(result, p)
        return result

File: 2025/day06_2/src/main/kotlin/CephalopodFormula.kt
package org.tlarsen.adventofcode2025.day06.part2

class CephalopodFormula {
    private val params = mutableListOf<Int>()
    fun addParam(newParam: Int) {
        params.add(newParam)
    }

    var operation : CephalopodOperation = CephalopodOperation.NONE

    override fun toString(): String {
        return params.joinToString(operation.toString())
    }

    fun doHomework() : Long = params.map { it.toLong() }.reduce { acc, i -> operation.calculate(acc, i) }
}